### PR TITLE
Add Matter firmware update documentation

### PR DESCRIPTION
**Pull request by agners**
## Proposed change

This adds preliminary documentation on how the Matter OTA will work in Home Assistant.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/120304
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced support for Over-the-Air (OTA) updates for Matter devices within Home Assistant.
	- Added a new section in the documentation detailing the functionality and usage of OTA updates for enhanced user experience.
	- Included a clear description of how Home Assistant checks for updates and how users can manually trigger update checks.

### CODE REVIEW

File: source/_integrations/matter.markdown [hunk 1]
```diff
@@ -318,6 +319,10 @@ This section provides a bit more information on some of the categories:
 
 **IP addresses**: Typically more than one IPv6 address is shown: link local, unique local, and global unicast. In some cases a device also supports IPv4. In that case there will also be listed an IPv4 address here.
 
+## Matter device updates
+
+The Matter standard supports OTA (Over-the-Air) updates optionally. Matter devices which support Matter updates will have an update entity. Furthermore, the CSA DCL (Distributed Compliance Ledger) stores firmware update information. Home Assistant reads firmware update information directly from the DCL to learn about available updates. By default the integration checks every 12 hours for an update. The `homeassistant.update_entity` service call with the update entity as target can be used to force an update check.
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
The Matter standard supports OTA (Over-the-Air) updates optionally. Matter devices that support Matter updates will have an update entity. Furthermore, the CSA DCL (Distributed Compliance Ledger) stores firmware update information. Home Assistant reads firmware update information directly from the DCL to learn about available updates. By default. the integration checks every 12 hours for an update. If you want to force an update check, use the `homeassistant.update_entity` [action](/integrations/homeassistant/) with the update entity as the target.
```

Comment (agners) [reply source/_integrations/matter.markdown]: Right! Thanks :pray:

**Review by c0ffeeca7 — DISMISSED** [review]

File: source/_integrations/matter.markdown [hunk 2]
```diff
@@ -318,6 +319,9 @@ This section provides a bit more information on some of the categories:
 
 **IP addresses**: Typically more than one IPv6 address is shown: link local, unique local, and global unicast. In some cases a device also supports IPv4. In that case there will also be listed an IPv4 address here.
 
+## Matter device updates
+
+The Matter standard supports OTA (Over-the-Air) updates optionally. Matter devices that support Matter updates will have an update entity. Furthermore, the CSA DCL (Distributed Compliance Ledger) stores firmware update information. Home Assistant reads firmware update information directly from the DCL to learn about available updates. By default, the integration checks every 12 hours for an update. If you want to force an update check, use the `homeassistant.update_entity` [action](/integrations/homeassistant/) with the update entity as the target.
```
Comment (frenck) [inline source/_integrations/matter.markdown]:
```suggestion
The Matter standard supports OTA (Over-the-Air) updates optionally. Matter devices that support Matter updates will have an [update entity](/integrations/update). Furthermore, the CSA DCL (Distributed Compliance Ledger) stores firmware update information. Home Assistant reads firmware update information directly from the DCL to learn about available updates. By default, the integration checks every 12 hours for an update. If you want to force an update check, use the `homeassistant.update_entity` [action](/integrations/homeassistant/) with the update entity as the target.

```

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request merged**